lib/main.coffee

```diff
@@ -0,0 +1,51 @@
+{CompositeDisposable} = require 'atom'
+
+module.exports =
+  config:
+    shellcheckExecutablePath:
+      type: 'string'
+      title: 'Shellcheck Executable Path'
+      default: 'shellcheck' # Let OS's $PATH handle the rest
+    enableNotice:
+      type: 'boolean'
+      title: 'Enable Notice Messages'
+      default: false
+
+  activate: ->
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe \
+     'linter-shellcheck.shellcheckExecutablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+    @subscriptions.add atom.config.observe 'linter-shellcheck.enableNotice',
+      (enableNotice) =>
+        @enableNotice = enableNotice
+
+  deactivate: ->
+    @subscriptions.dispose()
+
+  provideLinter: ->
+    helpers = require('atom-linter')
+    provider =
+      grammarScopes: ['source.shell']
+      scope: 'file'
+      lintOnFly: true
```

Comment:
`lintOnFly` is not about performance, it is meant for the linters that can lint a text buffer instead of a file, in case of `shellcheck`, You are providing it a path to file, so it'll lint what's saved on the disk, so invoking it on user change non-save events would be pointless :)